eventsourcing/src/main/java/org/axonframework/eventsourcing/eventstore/DefaultEventStoreTransaction.java

```diff
@@ -134,7 +134,7 @@ public void onAppend(@Nonnull Consumer<EventMessage<?>> callback) {
     }
 
     @Override
-    public long appendPosition(@Nonnull ProcessingContext context) {
-        return getOrDefault(context.getResource(appendPositionKey), -1L);
+    public ConsistencyMarker appendPosition(@Nonnull ProcessingContext context) {
```

Comment:
While @smcvb was explaining the internals of this class to me, we noticed a potential issue with AggregateBasedConsistencyMarker in this place. The appendPosition was primarily returned to ensure that your query model is up to date after processing a previously executed command. This works well with a global index, but if I only have the position in the aggregate stream (not the global one), it cannot fulfill this purpose.

Reply:
>Even though that might not be supported with this PR, making it an interface allows us to add support for this in other PRs.

And I think we should! :raised_hands: 
I anything, before this change, I feel #1740 was covered. With this change, that's no longer the case. 

Hence, I'll move the issue back on the backlog as a result of this.

Reply:
Yes, indeed. We could (at some point) expose an (Optional)Long on the consistency marker that gives an estimated position (not all implementations may be able to do it accurately) in the stream. If the position from a tracking token is beyond this one (and the message source of the event processor is this event store), then one may assume it has processed the events.